Ensure tree refreshes after sign in

RMacfarlane opened the pull request:
Follow up on https://github.com/microsoft/vscode-pull-request-github/issues/1269

With both the new 'sign in' option in the tree, and the sign in command, the tree was not properly updating after finishing auth. The problem was that within `updateRepositories`, we await for user response to a notification. I removed this call from that code path. Now, when the PR manager is initialized, it will show the notification if not logged in, but not block on it.

I also introduced an "Initialized" flag for the PR manager to indicate that a complete cycle of `updateRepositories` has been finished. This way, the tree will show "Loading..." until the repository data is completely there.

rebornix commented:
I think the tests fail as the tree refreshes and it's expecting the wrong tree items.

RMacfarlane commented:
Nice, I like the idea of the state enum a lot! I've updated to using that and then having the tree listen for changes to it. The listener for changes to the keychain is still there so that review manager state also updates then.

Fixed the tests, needed to stub out a method on the credential store.

There are still a couple of cases where the tree shows something that's not ideal. On startup, I don't think there's a way to get around the "no data provider registered" if we want the viewlet to always be there. In that case the viewlet has been restored from a cache but the extension hasn't activated yet so the tree data provider hasn't been registered. If we added a context key to the viewlet for extension activation, I think it would just cause it to disappear and then reappear on startup.

There's also still the issue on windows where `repositories` is initially an empty array, so I still see "No GitHub repositories found." before Git sends another update.

rebornix commented on src/github/pullRequestManager.ts:
```diff
@@ -494,11 +514,14 @@ export class PullRequestManager implements vscode.Disposable {
 		const allGitHubRemotes = await this.getAllGitHubRemotes();
```
@RMacfarlane this is the only place weird to me that we `await` for a non-async method.

RMacfarlane commented on src/github/pullRequestManager.ts:
```diff
@@ -114,7 +109,8 @@ export class PullRequestManager implements vscode.Disposable {
 	readonly onDidChangeActivePullRequest: vscode.Event<void> = this._onDidChangeActivePullRequest.event;
 
-	public state: UpdateRepositoryState;
+	public hasAuthenticated: boolean;
```
Thanks! I had missed updating this
